app/controllers/restaurants_controller.rb

```diff
@@ -13,6 +14,36 @@ def new
     @restaurant = Restaurant.new
   end
 
+  def edit
+    @restaurant = Restaurant.find(params[:id])
+    if @restaurant.owner?(current_user)
+      @restaurant
+    else
+      flash[:notice] = "Your account is not authorized to view this page"
+    end
+  end
```

Comment:
I think you're giving too much information to the user here. Why not just not let them see the option to edit in the view?